linera-witty/tests/wit_load.rs

```diff
@@ -353,6 +354,52 @@ fn test_heap_allocated_fields() {
     );
 }
 
+/// Check that a [`Vec`] type is properly loaded from memory and lifted from its flat
+/// layout.
+#[test]
+fn test_vec() {
+    let expected = vec![SimpleWrapper(false), SimpleWrapper(true)];
+
+    test_load_from_memory(&[8, 0, 0, 0, 2, 0, 0, 0, 0, 1], expected.clone());
```

Comment:
Why 8, 2 and 1? Would this have been different without this change?
(Sorry, I'm still not familiar enough with the inner workings of Witty. :sweat_smile:)

Reply:
The vec is stored in memory as a tuple of address and size (each four bytes), and the contents are stored right after the tuple. That means:

- 0..4: address (8, right after the tuple)
- 4..8: size (2)
- 8: `SimpleWrapper(false)` (0)
- 9: `SimpleWrapper(true)` (1)